File: Leetcode-Solutions/Solutions-Python/offer/maxTreeDepth.py
# -*- coding: utf-8 -*-
# @Time : 2020/7/14 8:50
# @Author : Kevin
# @IDE: PyCharm
# @ModuleName : maxTreeDepth
# @Description: 二叉树最大深度


# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    # DFS：左右子树最大深度+1
    def maxDepth(self, root: TreeNode) -> int:
        # 终止条件：root为空，越过叶子节点
        if not root: return 0
        # 递推：分别求左右子树的最大深度
        return max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1

File: Leetcode-Solutions/Solutions-Golang/offer/maxTreeDepth.go
package offer

//求树的最大深度

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {
	//终止条件：root为空，越过叶子节点
	if root == nil {
		return 0
	}
	//递推：分别求左右子树的最大深度
	return max(maxDepth(root.Left), maxDepth(root.Right)) + 1
}

func max(i, j int) int {
	if i > j {
		return i
	}
	return j
}
